rs/execution_environment/tests/dts.rs

```diff
@@ -3165,3 +3165,138 @@ fn yield_for_dirty_pages_copy_works_for_many_canisters() {
     // Only the first message per scheduler core must be completed after two rounds.
     assert_eq!(num_completed(), scheduler_cores);
 }
+
+#[cfg(not(all(target_arch = "aarch64", target_vendor = "apple")))]
+const WRITE_MORE_THAN_1G_ON_INIT_WAT: &str = r#"
+(module
+    (func (export "canister_init")
+        (local $i i32)
+        (local.set $i (i32.const 1073745920)) ;; 1GiB + 4096
+        (loop $loop
+            (i32.store (local.get $i) (i32.const 1))
+            (br_if $loop (local.tee $i (i32.sub (local.get $i) (i32.const 4096))))
+        )
+    )
+    (memory 16385) ;; 1GiB + 65536
+)"#;
+
+#[test]
+#[cfg(not(all(target_arch = "aarch64", target_vendor = "apple")))]
+fn yield_for_dirty_pages_copy_works_for_install_code() {
```

Comment:
IIUC, the problem would manifest even in case of long install code due to many instructions used, followed up by another install code immediately. Should the tests be a bit more general to cover that instead of focusing on the case of too many dirty pages touched? Is my understanding wrong?

Reply:
I added a test that executes 2.5 billion instructions during installation on top.